File: app/src/main/java/com/hmproductions/theredstreet/dagger/SharedPreferencesModule.java
package com.hmproductions.theredstreet.dagger;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import dagger.Module;
import dagger.Provides;

@Module (includes = ContextModule.class)
public class SharedPreferencesModule {

    @Provides
    @DalalStreetApplicationScope
    SharedPreferences getSharedpreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }
}

File: app/src/main/java/org/pragyan/dalal18/dagger/SharedPreferencesModule.kt
package org.pragyan.dalal18.dagger

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import androidx.security.crypto.EncryptedSharedPreferences
import androidx.security.crypto.MasterKey

import dagger.Module
import dagger.Provides
import org.pragyan.dalal18.utils.Constants

@Module
class SharedPreferencesModule {

    @Provides
    @DalalStreetApplicationScope
    fun getSharedPreferences(context: Context): SharedPreferences {
        val masterKey = MasterKey.Builder(context)
                .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
                .build()
        return EncryptedSharedPreferences.create(
                context,
                Constants.ENCRYPTED_SHARED_PREF_NAME,
                masterKey,
                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
        )
    }
}
